### ISSUE TITLE
Document kubelet --protect-kernel-defaults

### ISSUE THREAD

**Issue by YanzhaoLi** [kubernetes/kubernetes#107597]
### What happened?

There are 6 kernel parameters related to this parameters

- if not setting `--protect-kernel-defaults`: kubelet would restore the 6 kernel parameters to their default values when bootstrap, no matter what their values were.
- if setting `--protect-kernel-defaults`: kubelet would check whether the 6 kernel parameters were set to their default values, if not, kubelet would fail to start.

Thus, in both two cases, the 6 kernel parameters would be their default values when kubelet start.

### What did you expect to happen?

If in both cases the 6 parameters would be the same, I wonder what's the meaning of this parameters?  
Is it to let sysadmin know that kubelet has set the 6 kernel parameters to the default values?

### How can we reproduce it (as minimally and precisely as possible)?

N/A

### Kubernetes version

N/A

### Cloud provider

N/A

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>

### Install tools

<details>

</details>

### Container runtime (CRI) and and version (if applicable)

<details>

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

**Comment by yangjunmyfm192085** [comment]
Please refer here  `https://github.com/kubernetes/kubernetes/blob/018288b38fcf4c5248c83aa2920641b565798be2/pkg/kubelet/cm/container_manager_linux.go#L420`
Its meaning, kubelet needs to set these kernel parameters to run.
If you set the `--protect-kernel-defaults` flag, but kubelet finds that some parameters are inconsistent with the running requirements, it will report an error and  fail to start.
Can this be explained clearly?

**Comment by MadhavJivrajani** [comment]
/cc @sftim 
could this be a docs improvement maybe?

**Comment by sftim** [comment]
/uncc

This looks like a general question rather than one for me specifically.

**Comment by YanzhaoLi** [comment]
@yangjunmyfm192085  Yes.  But no matter whether enabling this flag or not, these 6 kernel parameters would be the same in both cases, as long as kubelet wants to run successfully.

So can I say that the only difference is to let sysadmin to know that they should manually set these 6 kernel parameters to the required values?  Because, disabling the flag, kubelet automatically sets these 6 kernel parameters to the same values.

**Comment by sftim** [comment]
I notice that the command line option `--protect-kernel-defaults` is deprecated. I don't think it's important to document it further. We should document the config file equivalent, though.

**Comment by endocrimes** [comment]
/triage accepted

A docs improvement for the config file would be great.

**Comment by sftim** [comment]
/retitle Document `--protect-kernel-defaults` argument to kubelet
